iwyu.cc

```diff
@@ -3000,8 +2983,27 @@ class InstantiatedTemplateVisitor
   bool TraverseTemplateSpecializationTypeHelper(
       const clang::TemplateSpecializationType* type) {
     if (CanIgnoreCurrentASTNode())  return true;
-    if (CanForwardDeclareType(current_ast_node()))
-      current_ast_node()->set_in_forward_declare_context(true);
+
+    ASTNode* const ast_node = current_ast_node();
+
+    // If, in this context, the template type is only a qualifier for a method
+    // defined out of line, don't treat it as an actual specialization. Just
+    // ignore it.
+    //
+    // Class<T,U>::method() {
+    // |-Type---^
+    // |-NNS------^
+    // |-CXXMethodDecl--^
+    //
+    if (auto nns = ast_node->GetParentAs<NestedNameSpecifier>())
```

Comment:
Please use braces around all nested ifs (when an inner if-block is braced, the freehanging outer ones look weird)